govcd/metadata_v2.go

```diff
@@ -954,43 +978,55 @@ func filterMetadata(allMetadata *types.Metadata, href, objectName string, metada
 
 	var filteredMetadata []*types.MetadataEntry
 	for _, originalEntry := range allMetadata.MetadataEntry {
-		_, err := filterSingleMetadataEntry(originalEntry.Key, href, objectName, &types.MetadataValue{Domain: originalEntry.Domain, TypedValue: originalEntry.TypedValue}, metadataToIgnore)
-		if err == nil || !strings.Contains(err.Error(), "ignored") {
-			filteredMetadata = append(filteredMetadata, originalEntry)
+		_, err := filterSingleXmlMetadataEntry(originalEntry.Key, href, objectName, &types.MetadataValue{Domain: originalEntry.Domain, TypedValue: originalEntry.TypedValue}, metadataToIgnore)
+		if err != nil {
+			if strings.Contains(err.Error(), "ignored") {
+				continue
+			}
+			return nil, err
 		}
+		filteredMetadata = append(filteredMetadata, originalEntry)
 	}
 	result.MetadataEntry = filteredMetadata
 	return result, nil
 }
 
-// filterSingleMetadataEntry filters a single metadata entry given a slice of metadata that needs to be ignored. It doesn't
+func filterSingleXmlMetadataEntry(key, href, objectName string, metadataEntry *types.MetadataValue, metadataToIgnore []IgnoredMetadata) (*types.MetadataValue, error) {
+	normalisedEntry, err := normaliseXmlMetadata(key, href, objectName, metadataEntry)
+	if err != nil {
+		return nil, err
+	}
+	isFiltered := filterSingleGenericMetadataEntry(normalisedEntry, metadataToIgnore)
+	if isFiltered {
+		return nil, fmt.Errorf("the metadata entry with key '%s' and value '%v' is being ignored", key, metadataEntry.TypedValue.Value)
+	}
+	return metadataEntry, nil
+}
+
+// filterSingleGenericMetadataEntry filters a single metadata entry given a slice of metadata that needs to be ignored. It doesn't
 // alter the input metadata, but returns a copy of the filtered metadata.
```

Comment:
This comment does not look exactly valid anymore as it returns `bool`, not a copy of filtered metadata.